standard-wallet-backend/README.md

```diff
@@ -0,0 +1,201 @@
+
+# **Abstract**
+
+Currently each light wallet uses a different backend and most of them do not offer a way to configure the backend host. Each backend exposing a different API makes it really difficult to be able to configure the backend host into the wallet settings.
+Being able to configure the backend host is really important to connect a wallet to a different Cardano network, so anyone can deploy a backend for the new network and connect the wallet to it.
+
+# **Motivation**
+
+As the Cardano ecosystem grows, more Cardano networks start to appear. Leading a project called Shareslake, I deployed a Cardano mainnet, finding that allowing people to connect existing wallets to this new network is really complex.
+
+To illustrate the problem let’s use Nami wallet as example. It allows to configure the transaction submit endpoint, nevertheless, it uses Blockfrost to read on-chain data such us the funds on the wallet address. So, even though you can change the submit endpoint, the wallet won’t show the address funds or transactions to the user.
+
+This CIP is created to suggest a standard on how wallets obtain the on-chain information from the backend, allowing people to easily edit the light wallet connection settings to connect it to a different Cardano network.
+What we are looking to achieve with this CIP is already possible into the Ethereum ecosystem. Let's think on a wallet called Metamask that you can easily connect to different Ethereum networks.
+
+# **Rationale**
+
+Focusing this CIP on the communication between backends indexing the ledger and the light wallets we should be able to suggest a bunch of standard endpoints that allows light wallet's users to just change the backend host and the creators of a new network to deploy a backend knowing that it will work with most of the light wallets.
+
+Cardano mainnet addresses do not include the network magic allowing to use the same address and keys between Cardano networks, so just changing of the backend host on the wallet settings is enough. Testnet addresses are different, they include the network magic, requiring to create a new address per network.
+
+There are two different points to standardise here:
+
+1. Endpoints used to read the on-chain data from the backend.
+2. Endpoints used to submit transactions to the network.
+
+The proposal must be agnostic to the wallet and backend implementations or programming languages so we will be defining REST endpoints that will be exposed by the backends and consumed by the wallets.
+
+To define those endpoints using [OpenAPI](https://swagger.io/specification/) seems ideal as it is widely used to specify RESTful APIs and there are several tools for automatically generating APIs code.
+
+About the first point, we could follow `cardano-db-sync` REST endpoints as standard for consuming on-chain data, but they were deprecated.
```

Comment:
You may be interested in Carp. It's a replacement for db-sync that also contains an openapi definition for rest endpoints for webclients. 

However, Carp doesn't handle ledger state (ex: reward balance) or off-chain information (pool metadata), so it will need to be combined in some kind of Cardano OneBox with other tools like Ogmios to provide a complete API that can easily be used for other deployments of Cardano. This is something we're interested in at dcSpark and I think TxPipe is also interested in this.

Reply:
I didn't know about those tools. I will take a look, if they are open-source maybe we could package all of them creating an easy way for deploying the backend.